src/components/common/Navbar/Navbar.tsx

```diff
@@ -42,30 +42,51 @@ function NavLinks({ onClickLink }: NavLinksProps) {
 
 function Navbar() {
   const [showMenu, setShowMenu] = useState(false)
+  const [searchExpanded, setSearchExpanded] = useState(false)
   const dismissTransition = useRef<Callback | undefined>()
   const handleCloseSlideOver = () => setShowMenu(false)
 
   return (
     <header className="navbar / grid-content-full">
       <div className="navbar__header / grid-content">
         <section className="navbar__row">
-          <IconButton
-            classes="navbar__menu"
-            aria-label="Open Menu"
-            icon={<ListIcon size={32} />}
-            onClick={() => setShowMenu(true)}
-          />
-          <LinkGatsby
-            to="/"
-            aria-label="Go to Faststore home"
-            title="Go to Faststore home"
-            className="navbar__logo"
-          >
-            <Logo />
-          </LinkGatsby>
+          {!searchExpanded && (
+            <>
+              <IconButton
+                classes="navbar__menu"
+                aria-label="Open Menu"
+                icon={<ListIcon size={32} />}
+                onClick={() => setShowMenu(true)}
+              />
+              <LinkGatsby
+                to="/"
+                aria-label="Go to Faststore home"
+                title="Go to Faststore home"
+                className="navbar__logo"
+              >
+                <Logo />
+              </LinkGatsby>
+            </>
+          )}
           <SearchInput />
-          <div className="navbar__buttons">
-            <SignInLink />
+          <div
+            className="navbar__buttons"
+            data-store-search-expanded={searchExpanded}
+          >
+            {searchExpanded && (
+              <IconButton
+                classes="retract__button"
```

Comment:
just to follow the pattern, It should be `navbar__something`, I would suggest `navbar__collapse`.  Does it make sense? 🤔